src/compressed_tensors/utils/offload.py

```diff
@@ -171,22 +171,24 @@ def update_parameter_data(
 
 def get_execution_device(module: torch.nn.Module) -> torch.device:
     """
-    Get the device which inputs should be moved to before module execution
+    Get the device which inputs should be moved to before module execution.
+    Assume that modules execute in the same order as returned by `model.modules()`
 
     :param module: module to check, may be offloaded
     :return: onload device of module
     """
-    if has_offloaded_params(module):
-        return module._hf_hook.execution_device
+    for module in module.modules():
+        if has_offloaded_params(module):
+            return module._hf_hook.execution_device
 
-    first_param = next(module.parameters(), None)
-    if first_param is None:
-        warnings.warn(
-            f"Unable able to infer execution device of {module}, falling back to CPU"
-        )
-        return torch.device("cpu")
+        param = next(module.parameters(recurse=False), None)
+        if param is not None:
+            return param.device
 
-    return first_param.device
+    warnings.warn(
+        f"Unable able to infer execution device of {module}, falling back to CPU"
```

Comment:
nit to avoid naming confusion
```suggestion
        f"Unable able to get execution device of {module}, falling back to CPU"
```